client/src/partials/primary_cash/income/transfer/transfer.html

```diff
@@ -20,44 +20,65 @@
           <i class="glyphicon glyphicon-share-alt"></i> {{ "TRANSFER.TRANSFER" | translate }}
         </div>
         <div class="panel-body">
-          <form ng-submit="submit()">
+          <form name="TransferForm" ng-submit="TransferCtrl.submit()" novalidate>
             <div class="form-group">
-              <label>{{ 'TRANSFER.PROJECT' | translate }}</label>
-              <select class="form-bhima" name="projectname" id="projectID" ng-model="data.project_id" ng-options="project.id as project.name for project in projects.data">
+              <label class="required">{{ 'TRANSFER.PROJECT' | translate }}</label>
+              <select
+                ng-model="TransferCtrl.data.project_id"
+                ng-options="project.id as project.name for project in TransferCtrl.projects.data"
+                class="form-bhima"
+                name="projectId"
+                required>
                 <option value="" disabled="disabled">-- {{ 'SELECT.PROJECT' | translate }} -- </option>
               </select>
             </div>
 
             <div class="form-group">
-              <label>{{ 'TRANSFER.CAISSE' | translate}}</label>
-              <!-- <select class="form-bhima" name="caissename" id="caisseID" ng-model="data.cash_box_id" ng-options="cash.id as cash.text for cash in cash_boxes.data">
-                <option value="" disabled="disabled">-- {{ 'SELECT.CASH' | translate }} -- </option>
-              </select> -->
-
-              <select class="form-bhima" ng-model="data.cash_box_id" required>
-                  <option value="" disabled="disabled">--- {{ 'SELECT.CASH' | translate }} ---</option>
-                  <option ng-repeat="cash in cash_boxes.data" ng-disabled="cash.is_auxillary == 0 && cash.is_bank == null" value="{{cash.id}}">
-                    {{ cash.text }}
-                  </option>
-                </select>
+              <label class="required">{{ 'TRANSFER.CAISSE' | translate}}</label>
+              <select
+                ng-model="TransferCtrl.data.cash_box_id"
+                class="form-bhima"
+                name="cashBoxId"
+                required>
+                <option value="" disabled="disabled">--- {{ 'SELECT.CASH' | translate }} ---</option>
+                <!-- FIXME ng-options is optimale but in this case we need to disable some options -->
+                <option ng-repeat="cash in TransferCtrl.cash_boxes.data"
```

Comment:
You can use `ng-options` here.  You can disable select options via a special syntax.

```html
<select ng-options="label disable when disable for value in array track by trackexpr"></select>
```

[Here is the documentation](https://docs.angularjs.org/api/ng/directive/ngOptions)